src/AbstractUri.php

```diff
@@ -448,15 +445,15 @@ private function formatRegisteredName(string $host)
 
         static $gen_delims = '/[:\/?#\[\]@ ]/'; // Also includes space.
         if (preg_match($gen_delims, $formatted_host)) {
-            throw new UriException(sprintf('the submitted host %s can not contain URI delimiters or a space', $host));
+            throw new UriException(sprintf('the submitted host `%s` is invalid : a registered name can not contain URI delimiters or spaces', $host));
         }
 
         $formatted_host = idn_to_ascii($formatted_host, 0, INTL_IDNA_VARIANT_UTS46, $arr);
         if (!$arr['errors']) {
             return $formatted_host;
         }
 
-        throw new UriException(sprintf('Host %s is invalid : %s', $host, $this->getIDNAErrors($arr['errors'])));
+        throw new UriException(sprintf('the submitted host `%s` is invalid : %s', $host, $this->getIDNAErrors($arr['errors'])));
```

Comment:
IMO the previous exception message was better with just `Host `%s` is invalid: %s`

Reply:
I can revert that ... I don't mind at all